pandas/core/dtypes/common.py

```diff
@@ -1688,38 +1688,69 @@ def is_extension_type(arr):
     return False
 
 
-def is_extension_array_dtype(arr_or_dtype):
-    """Check if an object is a pandas extension array type.
+def is_extension_array_dtype(arr):
+    """Check if an array object is a pandas extension array type.
 
     Parameters
     ----------
-    arr_or_dtype : object
+    arr : object
 
     Returns
     -------
     bool
 
     Notes
     -----
-    This checks whether an object implements the pandas extension
+    This checks whether an array object implements the pandas extension
     array interface. In pandas, this includes:
 
     * Categorical
+    * Interval
 
-    Third-party libraries may implement arrays or types satisfying
+    Third-party libraries may implement arrays satisfying
     this interface as well.
-    """
-    from pandas.core.arrays import ExtensionArray
 
-    if isinstance(arr_or_dtype, (ABCIndexClass, ABCSeries)):
-        arr_or_dtype = arr_or_dtype._values
+    See Also
+    --------
+    is_extension_dtype : Similar method for dtypes.
+    """
+    from pandas.core.dtypes.base import ExtensionDtype
 
     try:
-        arr_or_dtype = pandas_dtype(arr_or_dtype)
-    except TypeError:
-        pass
+        dtype = getattr(arr, 'dtype')
+    except AttributeError:
+        return False
+
+    return isinstance(dtype, ExtensionDtype)
 
-    return isinstance(arr_or_dtype, (ExtensionDtype, ExtensionArray))
+
+def is_extension_dtype(dtype):
+    """Check if a dtype object is a pandas extension dtype.
```

Comment:
One question is whether we want to accept strings here as well? 
Or keep it strictly for actual dtype objects?

Reply:
That's a good point, I suspect we should accept strings that have been registered.

I'll see what I can do.

Reply:
meant for the comment here
this is what pandas_dtype does